hyperdbg/hprdbgctrl/code/debugger/core/debugger.cpp

```diff
@@ -279,9 +279,9 @@ DebuggerGetNtoskrnlBase()
     UINT64               NtoskrnlBase = NULL;
     PRTL_PROCESS_MODULES Modules      = NULL;
 
-    Modules = (PRTL_PROCESS_MODULES)malloc(1024 * 1024);
+    Modules = (PRTL_PROCESS_MODULES)malloc(1024 * 1024 * sizeof(PRTL_PROCESS_MODULES));
 
-    NtQuerySystemInformation((SYSTEM_INFORMATION_CLASS)11, Modules, 1024 * 1024, NULL);
+    NtQuerySystemInformation((SYSTEM_INFORMATION_CLASS)11, Modules, 1024 * 1024 * sizeof(PRTL_PROCESS_MODULES), NULL);
```

Comment:
Please use the `STATUS_INFO_LENGTH_MISMATCH` and `STATUS_BUFFER_TOO_SMALL` errors to specify the buffer size.

```c
do 
{
    ...
    status = NtQuerySystemInformation(...);
    ...
} while (status == STATUS_INFO_LENGTH_MISMATCH);
```